qcodes_contrib_drivers/drivers/BlueFors/BlueFors.py

```diff
@@ -157,12 +157,17 @@ def get_temperature(self, channel: int) -> float:
 
             try:
                 # There is a space before the day for old BlueFors Control Sofware versions
-                df.index = pd.to_datetime(df['date']+'-'+df['time'], format=' %d-%m-%y-%H:%M:%S')
-            except:
+                date_time = pd.to_datetime(
+                    df["date"] + "-" + df["time"], format=" %d-%m-%y-%H:%M:%S"
+                )
+            except Exception:
                 # There is no space before the day with BlueFors Control Software v2.2
-                df.index = pd.to_datetime(df['date']+'-'+df['time'], format='%d-%m-%y-%H:%M:%S')
-
-            return df.iloc[-1]['y']
+                date_time = pd.to_datetime(
+                    df["date"] + "-" + df["time"], format="%d-%m-%y-%H:%M:%S"
+                )
+            df["date_time"] = date_time
+            df.set_index("date_time", inplace=True)
```

Comment:
since the index is discarded perhaps the intend here was to sort by index.

```suggestion
            df.set_index("date_time", inplace=True)
            df.sort_index(inplace=True)
```